source/includes/inkling-reference/_schemas.html.md

```diff
@@ -1,265 +1,480 @@
+
+For **dot range**, number of steps (numSteps) must be a positive integer.
 
-Note that start, stop, step, numSteps are all numeric literals.
+For **colon range**, the step size can be negative only if stop is less than start.
 
-‍
+For both **colon range** and **dot range**:
 
-###### Range Expression Rules
+* The **start point** is inclusive (it is included in the values of the range) and fixed. 
 
-There are three forms of range expressions which Inkling supports. 
-Select the Inkling tab to see an example of each type.
+* The **start point** is exact (to the maximum extent possible if the range expression type is floating point). 
+
+* The **end point** must be reachable from the range start by applying the step. (The range must be bounded.) 
+
+* The **end point** may or may not be included in the values of the range. If you
+land on it exactly it is in the range. If you don't land on it exactly it is not
+in the range. For example `Int8 {0:3:10}` gives you (0, 3, 6, 9). The end point
+`10` is not included. The specification of `10` as the stop point is not an error (because it is a limit, not an endpoint). 
+
+You can think of the **end point** as a limit. It is included only if, after
+applying the (step or numSteps) expression, you land on it exactly. 
+Otherwise the highest value landed on which is less than the end point is the final value in the range. 
+
+###### Value List Range Expressions
+
+Inkling supports range expressions for value lists. 
+ 
+> Value List Range Expression Syntax
+
+```inkling--syntax
+primitiveType 
+    '{' 
+        [ integerLiteral [ ',' integerLiteral ]* ]
+     |  [ floatLiteral [ ',' floatLiteral ]* ]
+     |  [ stringLiteral [ ',' stringLiteral ]* ]
+     |  [ booleanLiteral [ ',' booleanLiteral ]? ]
+    '}' 
+```
+
+A value list range
+expression can be defined for numeric types
+and the primitive types `Bool` and `String`.
+
+### Usage
 
 ```inkling--code
-# Example: Value list range expression
-schema Schema2
-  UInt8  {0,1,2,3,4}   num, # a set of UInt8 values
-  String {"a", "bc"}   cat  # a set of Strings
-end
+    String {"red", "3", "green"}    # is valid.
+    String {"red", 3, "green"}      # is invalid (integer value in string range).
+    UInt8  {7, 7, 7, 7}             # is valid.
+    UInt8  {7, -7, 7, 7}            # is invalid (negative integer in unsigned range).
 ```
 
-* **Value list range expression**
+Value list range expressions support allowing arbitrary lists of values to constitute the
+range expression set, rather than start and end points. 
+
+A value list range expression must have all its elements be valid values for the
+specified type. 
 
-A value list range expression is simply a list of values.
-These range expressions specify sets of values in 
-which each value is explicitly listed.
+A value list range expression must conform to the signed or unsigned attribute
+of the specified type (if there is such an attribute). 
+
+###### Schema Matching
```

Comment:
Does this section belong under "Types" or under "Schemas" in the outline?

Reply:
Schema matching is a type relationship so I think it belongs under Types (which is under Schemas).